### PR TITLE
fix: update meeting token if expired

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Screenshot 2024-10-14 at 11 13 57 PM

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

Discussion:- https://app.campsite.com/cal/posts/259mxravgwgl

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## Checklist

### CODE REVIEW

File: apps/web/lib/video/[uid]/getServerSideProps.ts [hunk 1]
```diff
@@ -78,10 +79,21 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
...
+  const fourteenDaysAfter = new Date(endTime.getTime() + 14 * 24 * 60 * 60 * 1000);
+  const epochTimeFourteenDaysAfter = Math.floor(fourteenDaysAfter.getTime() / 1000);
+
+  const videoReferencePassword = await updateMeetingTokenIfExpired({
+    bookingReferenceId: oldVideoReference.id,
+    roomName: oldVideoReference.uid,
+    meetingToken: oldVideoReference.meetingPassword,
+    exp: epochTimeFourteenDaysAfter,
+  });
+
```
Comment (Udit-takkar) [inline apps/web/lib/video/[uid]/getServerSideProps.ts:92]: If it has reached this part of code then current time must be less than booking.endTime + 14 days (buffer time)  because of this check 

  const isPast = booking?.endTime <= exitDate;
  if (isPast) {
    return {
      redirect: {
        destination: `/video/meeting-ended/${booking?.uid}`,
        permanent: false,
      },
    };
  }

File: packages/app-store/dailyvideo/lib/VideoApiAdapter.ts [hunk 2]
```diff
@@ -98,6 +98,44 @@ async function processTranscriptsInBatches(transcriptIds: Array<string>) {
...
+  exp,
+}: {
+  bookingReferenceId: number;
+  meetingToken: string;
+  roomName: string;
+  exp: number;
+}) => {
+  try {
+    await fetcher(`/meeting-tokens/${meetingToken}`).then(ZGetMeetingTokenResponseSchema.parse);
+  } catch (err) {
```
Comment (Udit-takkar) [inline packages/app-store/dailyvideo/lib/VideoApiAdapter.ts:114]: If there is a problem fetching meeting token then update the expiry time and save the new meeting token

**Review by zomars — APPROVED** [review]
@Udit-takkar can we follow up with unit tests for this edge case? Code LGTM

**Review by zomars — DISMISSED** [review]
We have a type error 🙏🏽 

image

**Review by zomars — APPROVED** [review]
